### PR TITLE
Add Gridicons to SelectDropdown

### PR DESCRIPTION
**Pull request by marekhrabe**
This PR adds an option to add Gridicon to SelectDropdown items. 

| Normal size | Compact |
| - | - |
| screen shot 2017-06-06 at 13 25 24 | screen shot 2017-06-06 at 13 25 15 |

New examples were added to devdocs so we can easily test and review this new addition: 

https://calypso.live/devdocs/design/select-dropdown?branch=add/icons-to-select-dropdown

### CODE REVIEW

**Comment by drw158** [comment]
Looks ok to me! Not positive on the icon sizing. In some popovers, we use 18px icons, in other places, 24px (add dropdown in editor). It'd be nice to be consistent everywhere, but I'm unsure on what to do.

@folletto do you have an opinion on that?

**Comment by folletto** [comment]
> In some popovers, we use 18px icons, in other places, 24px (add dropdown in editor). It'd be nice to be consistent everywhere, but I'm unsure on what to do.

Thanks for the ping. 
It's a good question, that I wouldn't have noticed myself. 

I did a quick review, and I'd keep this consistent with buttons, as it's styled and sized as buttons. Which means we should size them at `18px` both normal and compact.

File: client/components/select-dropdown/index.jsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import find from 'lodash/find';
 import filter from 'lodash/filter';
 import findIndex from 'lodash/findIndex';
 import map from 'lodash/map';
-import result from 'lodash/result';
+import get from 'lodash/get';
```
Comment (michaeldcain) [inline client/components/select-dropdown/index.jsx]: We can combine these into `import { find, filter, findIndex, get, map } from 'lodash';`

File: client/components/select-dropdown/index.jsx [hunk 2]
```diff
@@ -25,6 +26,7 @@ class SelectDropdown extends Component {
 	static propTypes = {
 		selectedText: PropTypes.string,
 		selectedCount: PropTypes.number,
+		selectedIcon: PropTypes.string,
```
Comment (michaeldcain) [inline client/components/select-dropdown/index.jsx]: Maybe add `icon` to `options` ~L41?

File: client/components/select-dropdown/index.jsx [hunk 3]
```diff
@@ -121,17 +123,18 @@ class SelectDropdown extends Component {
...
-		}
-
-		// return currently selected text
+		// return currently selected object
 		const selectedValue = selected ? selected : this.getInitialSelectedItem( this.props );
-		return result( find( options, { value: selectedValue } ), 'label' );
+		return find( options, { value: selectedValue } );
+	}
+
+	getSelectedText() {
```
Comment (michaeldcain) [inline client/components/select-dropdown/index.jsx]: Is `getSelectedText` still needed?

File: client/components/select-dropdown/README.md [hunk 4]
```diff
@@ -71,6 +75,10 @@ Optional number to show a Count next to the item text.
 
 ![selected example screenshot](https://cldup.com/BOZktaoqTT.png)
```
Comment (michaeldcain) [inline client/components/select-dropdown/README.md]: Can we add `disabled`?

File: client/components/select-dropdown/style.scss [hunk 5]
```diff
@@ -260,3 +260,14 @@ $compact-header-height: 35;
 		padding: 0px #{ $side-margin }px;
 	}
 }
+
+.select-dropdown__icon {
+	position: relative;
+	top: 7px;
+	margin-right: 8px;
```
Comment (michaeldcain) [inline client/components/select-dropdown/style.scss]: Maybe use the `$side-margin` variable divided by 2 for future updates.

File: client/components/select-dropdown/style.scss [hunk 6]
```diff
@@ -260,3 +260,14 @@ $compact-header-height: 35;
 		padding: 0px #{ $side-margin }px;
 	}
 }
+
+.select-dropdown__icon {
+	position: relative;
+	top: 7px;
```
Comment (jancavan) [inline client/components/select-dropdown/style.scss]: `top` needs to be indented.

**Comment by michaeldcain** [comment]
Closing - duplicate of #14682.

**Pull request closed without merging**